src/core/link.go

```diff
@@ -169,7 +169,7 @@ func (intf *link) handler() (chan struct{}, error) {
 		return nil, errors.New("failed to decode metadata")
 	}
 	if !meta.check() {
-		intf.links.core.log.Errorf("Failed to connect to node: %s is incompatible version (local %s, remote %s)",
+		intf.links.core.log.Debugf("Rejected incoming connection: %s is incompatible version (local %s, remote %s)",
```

Comment:
This shouldn't read "incoming", since it will display the same message for any connection (outgoing or incoming). I think "Rejected connection" would be good enough.